client/my-sites/stats/stats-date-control/index.tsx

```diff
@@ -102,10 +102,51 @@ const StatsDateControl = ( { slug, queryParams }: StatsDateControlProps ) => {
 		page( generateNewLink( shortcut.period, startDate, endDate ) );
 	};
 
+	const getShortcutForRange = () => {
+		const today = moment().format( 'YYYY-MM-DD' );
+		// Today
+		if ( today === dateRange.chartEnd && dateRange.daysInRange === 1 ) {
+			return shortcutList[ 0 ];
```

Comment:
This approach is hard-coding array indexes, so if we add or change anything, there is a high potential it will fail. Adding a loop and utilising offset and range from the shortcuts or even introducing a new property to the shortcut array would be less brittle.

Reply:
I did come back to this and added a loop for array traversal. I think the code is very context specific and can only be abstracted to a point. Each client of the date picker could potentially require special handling of their shortcuts too. If you have any thoughts on how it currently works (with the change to `Array.find()`, I'd love to look at this in more detail.